addons/l10n_it_edi/models/account_edi.py

```diff
@@ -0,0 +1,419 @@
+# -*- coding:utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models, fields, _
+from odoo.tests.common import Form
+from odoo.exceptions import UserError
+from odoo.tools import float_repr
+
+import re
+from datetime import date, datetime
+import logging
+import base64
+
+
+_logger = logging.getLogger(__name__)
+
+DEFAULT_FACTUR_ITALIAN_DATE_FORMAT = '%Y-%m-%d'
+
+
+class AccountEdi(models.Model):
+    _inherit = 'account.edi'
+
+    def _create_invoice_from_xml_tree(self, filename, tree):
+        self.ensure_one()
+        invoice = super()._create_invoice_from_xml_tree(filename, tree)
+        if invoice:
+            return invoice
+        if self.code != 'fattura_pa':
+            return invoice
+        if not re.search("([A-Z]{2}[A-Za-z0-9]{2,28}_[A-Za-z0-9]{0,5}.(xml.p7m|xml))", filename):
+            return invoice
+        else:
+            invoice = self.env['account.move'].create({})
+            return self._import_fattura_pa(tree, invoice)
+
+    def _update_invoice_from_xml_tree(self, filename, tree, invoice):
+        self.ensure_one()
+        if self.code != 'fattura_pa':
+            return super()._update_invoice_from_xml_tree(filename, tree, invoice)
+        if not re.search("([A-Z]{2}[A-Za-z0-9]{2,28}_[A-Za-z0-9]{0,5}.(xml.p7m|xml))", filename):
```

Comment:
Need a helper for this

Reply:
Also, something is missing here:
```
if multi:
  invoice.message_post("sorry but wa can't update your invoice because there is a lot of trucs in your xml")
  return False
```